xdem/spatial_tools.py

```diff
@@ -49,7 +51,7 @@ def get_array_and_mask(array: Union[np.ndarray, np.ma.masked_array], check_shape
         array = array.astype('float32')
 
     # Convert into a regular ndarray and convert invalid values to NaN
-    array_data = np.asarray(array).squeeze()
+    array_data = np.array(array).squeeze()
```

Comment:
Would that not make a copy of the array?
We should make sure that we do not copy the data when it's not needed.

Reply:
I can undo that, but I did it because values were modified inplace:
```python
ref_dem = ...  # a numpy array
tba_dem = ... # a numpy array
mask = ... # a numpy array

coreg.fit(ref_dem, tba_dem, mask=mask, transform=None) 

# Now ref_dem and tba_dem have been modified by the mask, since they are modified inplace.


```

Reply:
I don't think it's good practice to modify the inputs when they shouldn't be modified. Copying is the easiest way out. In the long term, we could maybe find out a way to ensure immutability without copying.

Reply:
Ok yes, we should certainly not modify the input DEMs in that case. But get_array_and_mask is now used many times in the code, and in some cases, it may not be wished to copy the data...
So maybe a copy after the call to get_array_and_mask should be favored? Or an option 'copy' in the function call?

Reply:
Copying is made if a masked_array is supplied, so it would be more consistent to copy by default... I see your point though that this is a potentially un-performant approach. We should have a benchmark system to validate the effects of this! In the tests, there's no apparent effect, but we do these operations on relatively small rasters..

Reply:
For now, I can add a `copy` argument, but say in parentheses that masked_arrays are always copied.

Reply:
But wait a minute, the `get_array_and_mask()` is currently very inconsistent in whether it will modify the input or not:

1. View if given a float array
2. Copy if given an integer array
3. View if given a float masked_array with `False` as its mask
4. Copy if given an integer array
4. Copy if given a float masked_array with a mask

Is this really what we want?? I get that it's good to return views as there is less memory duplication involved, but in this case, it really seems difficult to know if it's a view or a copy, no? What do you think, @rhugonnet?

Reply:
Nice spotting this, not good indeed. 
The copy argument seems like a good idea! Otherwise we could simply return views systematically, as those are more memory-efficient, and we copy the objects ourselves when it is needed in a certain processing chain?

Reply:
See my comment below. It's not possible to return views for arrays that do not accept NaNs...

Reply:
Added `copy` argument in 4f85f6b